hotel-management/src/types/form.ts

```diff
@@ -0,0 +1,39 @@
+import { ChangeEvent } from "react";
+
+export type FormErrors = Record<string, string | null>;
+
+export type Validator = (value: string, min?: number) => string | null;
+
+export type BaseRoomInput<T> = {
+  id: string;
+  name: string;
+  value: string;
+  label: string;
+  placeholder?: string;
+  validators?: Validator[];
+  onChange: (event: ChangeEvent<T>) => void;
+};
+
+export type RoomTextInput = BaseRoomInput<HTMLInputElement> & {
+  type: 'text' | 'number';
+};
+
+export type RoomSelectInput = BaseRoomInput<HTMLSelectElement> & {
+  type: 'select';
+  options: { label: string; value: string }[];
```

Comment:
Did I comment on this?
Can you help to check this PR https://github.com/TienNguyenDinh/training-react/pull/61